vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+`:h expression-syntax`
+
+### Numbers
+See `:h expr-number`
+
+```vim
+echo  123         |" Decimal
+echo  0b1111011   |" Binary
+echo  0173        |" Octal
+echo  0x7B        |" Hexadecimal
+echo  123.0       |" Floating-point
+echo  1.23e2      |" Floating-point (scientific notation)
+```
+
+Note that an *integer* number with a leading `0` is in octal notation. The
+usual arithmetic operations are supported.
+
+```vim
+echo  1 + 2       |" Addition
+echo  1 - 2       |" Subtraction
+echo  - 1         |" Negation (unary minus)
+echo  + 1         |" Unary plus (does nothing really, but still legal)
+echo  1 * 2       |" Multiplication
+echo  1 / 2       |" Division
+echo  1 % 2       |" Modulo (remainder)
+```
+
+### Booleans
+See `:h Boolean`
+
+The number 0 is false, every other number is true. Strings are implicitly
+converted to numbers (see below). There are two pre-defined semantic constants.
+
+```vim
+echo  v:true      |" Evaluates to 1 or the string 'v:true'
+echo  v:false     |" Evaluates to 0 or the string 'v:false'
+```
+
+Boolean values can result from comparison of two objects.
+
+```vim
+echo  1 == 2             |" Equality by value
+echo  1 != 2             |" Unequality
+echo  1 >  2             |" Greater than
+echo  1 >= 2             |" Greater than or equal
+echo  1 <  2             |" Smaller than
+echo  1 <= 2             |" Smaller than or equal
+echo  l1 is l2           |" Equality by identity (lists and dictionaries)
+
+" Regular expression
+echo  "hi" =~ "hello"    |" Regular expression match
+echo  "hi" !~ "hello"    |" Regular expression unmatch
+
+" For string comparison
+echo  "Hi" ==? "hi"      |" String equality, case-insensitive
+echo  "Hi" ==# "hi"      |" String equality, case-sensitive
+" and so on for other other inequality comparisons; comparison is done by
+" alphabetical order
+```
+
+Boolean operations are possible.
+
+```vim
+echo  v:true && v:false       |" Logical AND
+echo  v:true || v:false       |" Logical OR
+echo  ! v:true                |" Logical NOT
+echo  v:true ? 'yes' : 'no'   |" Ternary operator
+```
+
+
+### Strings
+See `:h String`
+
+An ordered zero-indexed sequence of bytes. The encoding of text into bytes
+depends on the option `:h 'encoding'`.
+
+```vim
+" Literal constructors
+echo  "Hello world\n"   |" The last two characters stand for newline
+echo  'Hello world\n'   |" The last two characters are literal
+echo  'Let''s go!'      |" Two single quotes become one quote character
+```
+
+Single-quote strings take all characters are literal, except two single quotes,
+which are taken to be a single quote in the string itself. See `:h expr-quote`
+for all possible escape sequences.
+
+```
+" String concatenation
+echo  'Hello ' .  'world'  |" String concatenation
+echo  'Hello ' .. 'world'  |" String concatenation
+
+" String indexing
+echo  'Hello'[0]           |" First byte
+echo  'Hello'[1]           |" Second byte
+echo  'Hello'[-1]          |" Last byte
```

Comment:
This doesn't work for me, `'hello'[-1:-1]` works, can you test it in vim?

Reply:
Right, that was a mistake.